frontend/src/infra/template/createTemplate.ts

```diff
@@ -0,0 +1,34 @@
+import { Template, TemplateIntent } from '@/domain/Template'
+import { ApiClient } from '../lib/apiClient'
+import { BaseError } from '@/_lib/errors/BaseError'
+
+const createTemplateError = (message = 'Failed to create template') => {
+  const name = 'CreateTemplateError'
+
+  return new BaseError({ message, name, code: name })
+}
+
+export const makeCreateTemplate = (apiClient: ApiClient) => async (template: TemplateIntent) => {
```

Comment:
It's a bit confusing that this is called `makeCreateTemplate` instead of `createTemplate` like we have the other infra methods: `createTask`, `deleteTask`, etc.

I would rename this one to `createTemplate` to keep the same pattern and rename the action `createTemplate` to `createTemplateAction`.

Reply:
I named it `makeCreateTemplate` because its a function that creates another function. I actually started with `createTemplateAction` but I thought it could be mistaken by "creating a template-action".

This pattern for naming is common for function factories in the js environment, you can even check an example on this DI container lib:

https://github.com/jeffijoe/awilix

```ts
// Let's try with a factory function.
const makeUserService = ({ db }) => {
  // Notice how we can use destructuring
  // to access dependencies
  return {
    getUser: id => {
      return db.query(`select * from users where id=${id}`)
    }
  }
}
```

Reply:
Check how it looks on the code:
```ts
  const apiClient = await serverFetch()
  const createTemplate = makeCreateTemplate(apiClient)
  ```
  
  the `createTemplate` function is the function that is returned by `makeCreateTemplate`

Reply:
Ok let's leave it like this then